### PR TITLE
Support for jit-ed block reorder on AArch64

### PR DESCRIPTION
**Pull request by cfRod**
Re-opening a new PR for commit https://github.com/tensorflow/tensorflow/pull/61123/commits/a071af0ca8a3ea1fe242094514507f649f3ab64e

### CODE REVIEW

**Comment by cfRod** [comment]
@penpornk

File: tensorflow/workspace2.bzl [hunk 1]
```diff
@@ -207,6 +207,8 @@ def _tf_repositories():
             "//third_party/mkl_dnn:onednn_acl_fixed_format_kernels.patch",
             "//third_party/mkl_dnn:onednn_acl_depthwise_convolution.patch",
             "//third_party/mkl_dnn:onednn_acl_threadpool_scheduler.patch",
+            "//third_party/mkl_dnn:onednn_reorder_padded.patch",
```
Comment (penpornk) [inline tensorflow/workspace2.bzl:210]: To be consistent to #61295,
```suggestion
            "//third_party/mkl_dnn:onednn_acl_reorder_padded.patch",
```

**Review by penpornk — CHANGES_REQUESTED** [review]

**Review by penpornk — CHANGES_REQUESTED** [review]
#61295 was merged. Could you please rebase? Thank you!

**Review by penpornk — APPROVED** [review]
Thank you! :)

File: third_party/mkl_dnn/onednn_acl_reorder_update.patch [hunk 2]
```diff
@@ -0,0 +1,4193 @@
...
++    const auto G = with_groups_ ? od.padded_dims()[0] : 1;
++    const auto N = od.padded_dims()[with_groups_ ? 1 : 0];
++    static constexpr int cache_line_size = 16;
++    const auto wspace_per_thr_size
++            = utils::rnd_up(G * N, cache_line_size) * sizeof(int32_t);
++
++    auto scratchpad = scratchpad_registry().registrar();
++    const auto compensation_reduce_size = wspace_per_thr_size * nthr_;
++
++    //every thread gets its own scratchpad space for each N
```
Comment (penpornk) [inline third_party/mkl_dnn/onednn_acl_reorder_update.patch:2511]: Nit:
```suggestion
+    // Every thread gets its own scratchpad space for each N
```

File: third_party/mkl_dnn/onednn_acl_reorder_update.patch [hunk 3]
```diff
@@ -0,0 +1,4193 @@
...
+-                    omp_driver_4d(ithr, nthr, ndims_ker, in, out, scale);
++                    omp_driver_4d(ithr, nthr, ndims_ker, in, out, scale, src_zp,
++                            dst_zp, compensation_scratch);
+                     break;
+                 default: assert(!"unimplemented");
+             }
+         });
+     }
++
++    //reduction of intermediate compensation results to the final output
```
Comment (penpornk) [inline third_party/mkl_dnn/onednn_acl_reorder_update.patch:2862]: Nit:
```suggestion
+    // Reduction of intermediate compensation results to the final output
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Minor fixes please. (They are failing our internal checks.)

**Comment by cfRod** [comment]
Done!

**Review by penpornk — APPROVED** [review]
Thank you! :)

**Pull request merged**